akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -268,6 +268,40 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Sliding[T](n: Int, step: Int) extends PushPullStage[T, immutable.Seq[T]] {
+  private var buf = immutable.Stack.empty[T]
```

Comment:
Indeed, Stack is constant cost, however not very fast in general. Vector may need to resize when appending stuff to it, that's why amortized constant cost. However in the situation we have here we can use a builder, with size hint, and get some excellent performance:

```
[info] Benchmark                           Mode  Cnt       Score       Error   Units
[info] TestBenchmark.stack_1000           thrpt   20       7.446 ±     0.338  ops/ms
[info] TestBenchmark.vector_1000          thrpt   20   19401.495 ±   754.851  ops/ms
[info] TestBenchmark.vector_builder_1000  thrpt   20  142296.432 ± 40669.008  ops/ms
```

https://gist.github.com/ktoso/6f1ed3dc192b24d93f97 (trust no-one, bench everything)